src/util/OrmUtils.ts

```diff
@@ -121,6 +121,26 @@ export class OrmUtils {
         return true;
     }
 
+    /**
+     * Check if two entity-id-maps are the same
+     */
+    static compareIds(firstId: ObjectLiteral|undefined, secondId: ObjectLiteral|undefined) {
+        if (firstId === undefined || firstId === null || secondId === undefined || secondId === null)
+            return false;
+
+        // Optimized version for the common case
+        if (
+            typeof firstId.id === "string" &&
```

Comment:
but what if ids are numbers? wouldn't it be an optimized version for numeric ids too?

Reply:
@pleerock agreed, I've pushed a new version which works for "string" and "number". wdyt?